Chessington.GameEngine/Pieces/Piece.cs

```diff
@@ -20,5 +20,46 @@ public void MoveTo(Board board, Square newSquare)
             var currentSquare = board.FindPiece(this);
             board.MovePiece(currentSquare, newSquare);
         }
+
+        protected IEnumerable<Square> GetAvailableLateralMoves(Board board)
+        {
+            var currentSquare = board.FindPiece(this);
+
+            var zeroes = Enumerable.Repeat(0, 15);
+            var minusSevenToSeven = Enumerable.Range(-7, 15);
+
+            var moveVectors = zeroes.Zip(minusSevenToSeven, (x, y) => new MoveVector(x,y));
+            moveVectors = moveVectors.Union(minusSevenToSeven.Zip(zeroes, (x, y) => new MoveVector(x, y)));
```

Comment:
I think something like this would be a bit more readable:
```
var minusSevenToSeven = ...
var xChangedMoveVectors = minusSevenToSeven.Select(x => new MoveVector(x, 0));
var yChangedMoveVectors = minusSevenToSeven.Select(y => new MoveVector(0, y));
var lateralMoveVectors = xChangedMoveVectors.Concat(yChangedMoveVectors);
```
What do you think? 🙂

Reply:
Yep, the use of `Zip` is much less readable than that, thank you! Sorted in d4877e8